### ISSUE TITLE
[java] - SingularField with Gauge from micrometer

### ISSUE THREAD

**Issue by samuel-casasdray** [pmd/pmd#5451]
**Affects PMD Version:** 

**Rule:**

Please provide the rule name and a link to the rule documentation: 
*Rule*: SingularField
*Link*: https://docs.pmd-code.org/latest/pmd_rules_java_design.html#singularfield

**Description:**

We get a false positive when we have a class with a simple variable, a setter and a gauge (in the constructor) from micrometer.
The gauge from multimeter show the current value of a meter.
And so yes as the variable is private without at first glance a use because there is no getter and only a setter for this, PMD assume that the variable don't need to be global and can totally be local, but the thing is, gauge monitor it and others can access it by the name given when created.
And we can't put it as local in either the setter or constructor because it's needed in both. In the constructor to create the gauge and in the setter to update the value.

**Code Sample demonstrating the issue:**

```java
public class Bar {
	private final MeterRegistry meterRegistry;
	private Integer foo = 0;

	public Bar(MeterRegistry meterRegistry) {
		this.meterRegistry = meterRegistry;
		Gauge.builder("gauge.foo", foo, Integer::doubleValue).register(meterRegistry);
	}

	public void enabledPairs(int foo) {
		this.foo = foo;
	}
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *Maven* and with PMD 7.7.0

P.S.: Maybe it shouldn't be a PMD issue and more a micrometer, in this case tell me and I will close this one and open it elsewhere
